javascript-bank/bank.js

```diff
@@ -0,0 +1,37 @@
+/* exported Bank */
+function Bank() {
+  this.accounts = [];
+  this.nextAccountNumber = 1;
+}
+
+Bank.prototype.openAccount = function (holder, balance) {
+  if ((balance > 0) && (balance % 1 === 0)) {
+    let acc = new Account(balance, holder);
+    acc.nextAccountNumber = this.nextAccountNumber;
+    acc.balance = balance;
```

Comment:
In line with "encapsulation", external code should not directly modify an `Account` instance's properties. It should make use of the `Account`'s methods instead.